client/src/app/core/core-services/model-request-builder.service.ts

```diff
@@ -83,12 +83,53 @@ interface BaseSimplifiedModelRequest {
     additionalFields?: AdditionalField[];
 }
 
+interface DescriptorResponse<T extends FieldDescriptor> {
+    descriptor: T;
+    collectionsToUpdate: Collection[];
+}
+
 export interface Fieldsets<M extends BaseModel> {
     [name: string]: (keyof M | IAllStructuredFields)[];
 }
 
 export const DEFAULT_FIELDSET = 'detail';
 
+class ModelRequestObject {
+    public readonly collection: Collection;
+    public readonly ids: Id[] | undefined;
+    public readonly fields: Fields;
+    public readonly simplifiedRequest: BaseSimplifiedModelRequest;
+    public readonly collectionsToUpdate: Collection[] = [];
```

Comment:
I had problems with this kind of initialization before. The array might be static, so every instance of this class might share this array. Please set it in the constructor.

Reply:
I don't know, if I understand you correctly, but I don't like to use a static array/set.